### PR TITLE
Change index interval, add slug to search documents

### PR DESCRIPTION
**Pull request by Geometrically**

### CODE REVIEW

File: src/routes/mods.rs [hunk 1]
```diff
@@ -133,13 +133,30 @@ pub async fn mod_slug_get(
...
+            .await
+            .map_err(|e| ApiError::DatabaseError(e.into()))?;
+
+        if mod_data.is_none() {
+            mod_data = database::models::Mod::get_full_from_slug(string, &**pool)
+                .await
+                .map_err(|e| ApiError::DatabaseError(e.into()))?;
+        }
+    } else {
+        mod_data = database::models::Mod::get_full_from_slug(string, &**pool)
```
[the remark this answers is not in the document]
Comment (Aeledfyr) [reply src/routes/mods.rs:155]: These routes should also document the format that they expect.  Are you planning on supporting `@` prefixing to allow distinguishing ids from slugs when they conflict?  That doesn't look to be implemented.

**Review by Aeledfyr — CHANGES_REQUESTED** [review]

File: src/routes/mods.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Geometrically) [reply src/routes/mods.rs:155]: The old route still exists. All this is doing is making so the root route also works with usernames. If they conflict, the `@` can still be used,

File: src/scheduler.rs [hunk 2]
```diff
@@ -36,8 +36,8 @@ pub fn schedule_versions(
     pool: sqlx::Pool<sqlx::Postgres>,
     skip_initial: bool,
 ) {
-    // Check mojang's versions every 6 hours
-    let version_index_interval = std::time::Duration::from_secs(60 * 60 * 6);
+    // Check mojang's versions every 10 minutes
+    let version_index_interval = std::time::Duration::from_secs(60 * 10);
```
[the remark this answers is not in the document]
Comment (Geometrically) [reply src/scheduler.rs:40]: Will make it configurable.

File: src/routes/users.rs [hunk 3]
```diff
@@ -192,11 +191,6 @@ pub async fn teams(
 #[derive(Serialize, Deserialize)]
 pub struct EditUser {
     pub username: Option<String>,
-    #[serde(
-        default,
-        skip_serializing_if = "Option::is_none",
-        with = "::serde_with::rust::double_option"
-    )]
```
Comment (Aeledfyr) [inline src/routes/users.rs:183]: Why was this removed without removing the double optional on name?  I don't think there's a valid way of entering a name now, without something like `{"name":{"Some":"username here"}}`

Comment (Geometrically) [reply src/routes/users.rs:183]: Oops

**Review by Aeledfyr — APPROVED** [review]
